src/main/java/com/couchbase/lite/Database.java

```diff
@@ -1092,17 +1120,48 @@ public synchronized void open() throws CouchbaseLiteException {
             throw new CouchbaseLiteException("Database directory is not directory",
                     Status.INTERNAL_SERVER_ERROR);
 
-        // Initialize & open store
-        store = createStoreInstance();
-        if(store == null)
-            store = new SQLiteStore(path, manager, this);
+        String storageType = options.getStorageType();
+        if (storageType == null)
+            storageType = manager.getStorageType();
+        if (storageType == null)
+            storageType = DEFAULT_STORAGE;
+
+        Store primaryStore = createStoreInstance(storageType);
+        if (primaryStore == null) {
+            if (storageType.equals(Manager.SQLITE_STORAGE) || storageType.equals(Manager.FORESTDB_STORAGE))
+                Log.w(TAG, "storageType is '%s' but no class implementation found", storageType);
+            throw new CouchbaseLiteException("Can't open database in that storage format",
+                    Status.INVALID_STORAGE_TYPE);
+        }
+
+        boolean primarySQLite = Manager.SQLITE_STORAGE.equals(storageType);
+        Store otherStore = createStoreInstance(primarySQLite ? Manager.FORESTDB_STORAGE : Manager.SQLITE_STORAGE);
```

Comment:
To be able to call otherStore.databaseExists(path) below. Other option would be making databaseExists() a static method and use reflection to call databaseExists() without instantiating an object. But it seems to be too much work.